shared/data/src/main/java/com/ivy/data/repository/impl/ExchangeRatesRepositoryImpl.kt

```diff
@@ -0,0 +1,105 @@
+package com.ivy.data.repository.impl
+
+import com.ivy.base.threading.DispatchersProvider
+import com.ivy.data.db.dao.read.ExchangeRatesDao
+import com.ivy.data.db.dao.write.WriteExchangeRatesDao
+import com.ivy.data.db.entity.ExchangeRateEntity
+import com.ivy.data.model.ExchangeRate
+import com.ivy.data.remote.RemoteExchangeRatesDataSource
+import com.ivy.data.remote.responses.ExchangeRatesResponse
+import com.ivy.data.repository.ExchangeRatesRepository
+import com.ivy.data.repository.mapper.ExchangeRateMapper
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class ExchangeRatesRepositoryImpl
+    @Inject
+    constructor(
+        private val mapper: ExchangeRateMapper,
+        private val exchangeRatesDao: ExchangeRatesDao,
+        private val writeExchangeRatesDao: WriteExchangeRatesDao,
+        private val remoteExchangeRatesDataSource: RemoteExchangeRatesDataSource,
+        private val dispatchersProvider: DispatchersProvider,
+    ) : ExchangeRatesRepository {
+        override val urls =
```

Comment:
This urla logic doesn't belong to the repository. It's a responsibility of the `RemoteExchangeRatesDataSource` so you should encapsulate it there